apps/balancer-tools/src/app/poolsimulator/(utils)/index.ts

```diff
@@ -26,45 +51,39 @@ export function convertAnalysisDataToAMM(data: AnalysisData) {
             priceRate: String(token.rate),
           })),
           tokensList: data.tokens.map((token) => String(token.symbol)),
-        }),
+        })
       );
     }
     case PoolTypeEnum.GyroE: {
-      return new AMM(
-        new ExtendedGyroEV2({
-          swapFee: String(data.poolParams?.swapFee),
-          totalShares: String(
-            data.tokens.reduce((acc, token) => acc + token.balance, 0),
-          ),
-          tokens: data.tokens.map((token) => ({
-            address: String(token.symbol), // math use address as key, but we will use symbol because custom token will not have address
-            balance: String(token.balance),
-            decimals: token.decimal,
-            priceRate: String(token.rate),
-          })),
-          tokensList: data.tokens.map((token) => String(token.symbol)),
+      return fetchECLPDerivativeParams(data)
+        .then((derivedParams) => {
```

Comment:
since this is an async function you can just await the fetch -- think we should commit to one pattern unless there's a specific reason for not doing it